sdk/metric/meter.go

```diff
@@ -26,6 +27,14 @@ import (
 	"go.opentelemetry.io/otel/sdk/metric/internal"
 )
 
+var (
+	instrumentNameRe = regexp.MustCompile(`^([A-Za-z]){1}([A-Za-z0-9\_\-\.]){0,62}$`)
+
+	// ErrInvalidInstrumentName indicates the created instrument has an invalid name.
+	// Valid names must consist of 63 or fewer characters including alphanumeric, _, ., -, and start with a letter.
+	ErrInvalidInstrumentName = errors.New("invalid instrument name")
```

Comment:
nit: both `Err` and `Invalid` identify this variable represents something wrong with the instrument name. The `Invalid` part can be dropped.

```suggestion
	ErrInstrumentName = errors.New("invalid instrument name")
```